bin/spice/cmd/upgrade.go

```diff
@@ -45,103 +48,203 @@ spice upgrade
 			return
 		}
 
-		slog.Info("Checking for latest Spice CLI release...")
-		release, err := github.GetLatestCliRelease()
-		if err != nil {
-			slog.Error("checking for latest release", "error", err)
-			return
-		}
-
 		rtcontext := context.NewContext()
 		err = rtcontext.Init()
 		if err != nil {
 			slog.Error("initializing runtime context", "error", err)
 			os.Exit(1)
 		}
-		cliVersion := version.Version()
+
+		if os.Getenv(constants.SpiceUpgradeReloadEnv) != "true" {
+			// Run CLI upgrade
+			if !upgradeCli(force, rtcontext) {
+				// Exit if CLI upgrade fail
+				return
+			}
+		}
+
+		// Cleanup old binaries on windows
+		if runtime.GOOS == "windows" {
```

Comment:
Nit: we can call this for all platforms so the logic is consistent and perform clean up on Windows only (what we already check/do)

```
func cleanupOldBinaries() {
	if !util.IsWindows() {
		return
	}
```